### ISSUE TITLE
[JENKINS-25912] Workspace remains on disk after job has been ( deleted, moved, renamed )

### ISSUE THREAD

**Issue by jenkins-infra-bot** [timja-org/jenkins-gh-issues-poc-11-23#12317]
<p>Scenario_1:</p>

<p>Job foo runs once and creates a workspace named "foo".<br/>
After renaming foo to foofoo the workspace "foo" still exists if it wasn't deleted manually.<br/>
If foofoo runs now it creates it's own workspace.</p>

<pre>
# ls 
foo 
foofoo
</pre>

<p>The problem by having multiple jobs with huge workspaces is the disk space. In addition to that it's not nice to have some dead folders lying around on your file-system.</p>

<p>Scenario_2:<br/>
If job foo is executed in parallel it creates multiple workspaces.<br/>
This is fine as long as you're not using the "Wipe Out Current Workspace" button. <br/>
If so the only workspace that is getting deleted is foo but not the additional ones.</p>

<p>Example <b>before</b> wiping the workspace:</p>

<pre>
#ls
foo
foo@​2
foo@​3
foo@​4
</pre>

<p>Example <b>after</b> wiping the workspace:</p>

<pre>
#ls

foo@​2
foo@​3
foo@​4
</pre>

---
<details><summary><i>Originally reported by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jthenne">jthenne</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-25912" target="_blank">Workspace remains on disk after job has been ( deleted, moved, renamed )</a></i></summary>
<i><ul>
<li><b>status</b>: Open
<li><b>priority</b>: Minor
<li><b>component(s)</b>: core
<li><b>label(s)</b>: workspace
<li><b>resolution</b>: Unresolved
<li><b>votes</b>: 2
<li><b>watchers</b>: 3
<li><b>imported</b>: 2025-11-23
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>Scenario_1:</p>

<p>Job foo runs once and creates a workspace named "foo".
After renaming foo to foofoo the workspace "foo" still exists if it wasn't deleted manually.
If foofoo runs now it creates it's own workspace.</p>

<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java"># ls 
foo 
foofoo
</pre>
</div></div>

<p>The problem by having multiple jobs with huge workspaces is the disk space. In addition to that it's not nice to have some dead folders lying around on your file-system.</p>

<p>Scenario_2:
If job foo is executed in parallel it creates multiple workspaces.
This is fine as long as you're not using the "Wipe Out Current Workspace" button. 
If so the only workspace that is getting deleted is foo but not the additional ones.</p>

<p>Example <b>before</b> wiping the workspace:</p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">#ls
foo
foo@2
foo@3
foo@4
</pre>
</div></div>

<p>Example <b>after</b> wiping the workspace:</p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">#ls

foo@2
foo@3
foo@4
</pre>
</div></div>

</pre>
</details>
</details>
<details><summary><i>environment</i></summary>

```
1.590<br/>
redhat 5<br/>
Java 7
```
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-25912?focusedId=217563&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-217563">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Straightforward workaround present (remove workspaces manually), therefore reducing priority.</p>

<p>Additional scenario: Roaming jobs probably also retain their older workspaces on other nodes when deleting the current one.</p>
</pre>
</details>

</ul>
</details>

<p>Straightforward workaround present (remove workspaces manually), therefore reducing priority.</p>

<p>Additional scenario: Roaming jobs probably also retain their older workspaces on other nodes when deleting the current one.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-25912?focusedId=217658&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-217658">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>The problem of course is offline nodes while performing the action. Those simply cannot be updated.</p>
</pre>
</details>

</ul>
</details>

<p>The problem of course is offline nodes while performing the action. Those simply cannot be updated.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/gggeek.jpg" title="gggeek's avatar" /> gggeek:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-25912?focusedId=230831&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-230831">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Removing workspaces dirs manually involves at least one shell script which checks if no corresponding jobs exist any more.
And, when slaves do exist, it is not as easy as that.
I wouldn't mind a plugin, but it should be available as a feature.</p>
</pre>
</details>

</ul>
</details>

<p>Removing workspaces dirs manually involves at least one shell script which checks if no corresponding jobs exist any more.<br/>
And, when slaves do exist, it is not as easy as that.<br/>
I wouldn't mind a plugin, but it should be available as a feature.</p>

**Comment by jenkins-infra-bot** [comment]
<i>[Original `is_duplicated_by` from Jira: <a href="https://github.com/timja-org/jenkins-gh-issues-poc-11-23/issues?q=is%3Aissue%20%22jira_issue_key%3DJENKINS-25967%22">JENKINS-25967</a>]</i>
